File: 0242-valid-anagram/0242-valid-anagram.cpp
#include<algorithm>
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }

        // std::sort(s.begin(), s.end());
        // std::sort(t.begin(), t.end());
        // return s == t;

        std::unordered_map<char, int> dict;

        for(auto& ch:s){
            dict[ch]++;
        }


        for(auto& ch: t){
            if(dict.find(ch) == dict.end()){
                return false;
            }
            dict[ch] -=1;
            if(dict[ch] == 0){
                 dict.erase(ch);

            }

        }

        return dict.empty();
        
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        # A = sorted(s)
        # B = sorted(t)

        # return A == B

        mapA = {}
        mapB = {}

        for ch in s:
            mapA[ch] = mapA.get(ch, 0)+1
        
        for ch in t:
            mapB[ch] = mapB.get(ch,0) +1


        

        
        
        print(mapA)

        return mapA == mapB
        